codegen/files/rest/server.go

```diff
@@ -1,11 +1,587 @@
 package rest
 
 import (
+	"fmt"
+	"text/template"
+
 	"goa.design/goa.v2/codegen"
+	"goa.design/goa.v2/codegen/files"
+	"goa.design/goa.v2/design"
 	"goa.design/goa.v2/design/rest"
+	"net/http"
+	"sort"
+	"strings"
+)
+
+//todo: in encode headers and mappings
+
+const serverStructT = `{{ printf "%s lists the %s service endpoint HTTP handlers." .HandlersStruct .ServiceName | comment }}
+type {{ .HandlersStruct }} struct {
+	{{- range .ActionData }}
+	{{ .EndpointName }} http.Handler
+	{{- end }}
+}
+`
+
+const serverConstructorT = `{{ printf "%s instantiates HTTP handlers for all the %s service endpoints." .Constructor .ServiceName | comment }}
+func {{ .Constructor }}(
+	e *endpoints.{{ .ServiceName }},
+	dec rest.RequestDecoderFunc,
+	enc rest.ResponseEncoderFunc,
+	logger goa.Logger,
+) *{{ .HandlersStruct }} {
+	return &{{ .HandlersStruct }}{
+		{{- range .ActionData }}
+		{{ .EndpointName }}: {{ .Constructor }}(e.{{ .EndpointName }}, dec, enc, logger),
+		{{- end }}
+	}
+}
+`
+
+const serverMountT = `{{ printf "%s configures the mux to serve the %s endpoints." .MountHandlers .ServiceName | comment }}
+func {{ .MountHandlers }}(mux rest.ServeMux, h *{{ .HandlersStruct }}) {
+	{{- range .ActionData }}
+	{{ .MountHandler }}(mux, h.{{ .EndpointName }})
+	{{- end }}
+}
+`
+
+const serverHandlerT = `{{ printf "%s configures the mux to serve the \"%s\" service \"%s\" endpoint." .MountHandler .ServiceName .EndpointName | comment }}
+func {{ .MountHandler }}(mux rest.ServeMux, h http.Handler) {
+	{{- range .Routes }}
+	mux.Handle("{{ .Method }}", "{{ .Path }}", h)
+	{{- end }}
+}
+`
+
+const serverHandlerConstructorT = `{{ printf "%s creates a HTTP handler which loads the HTTP request and calls the \"%s\" service \"%s\" endpoint." .Constructor .ServiceName .EndpointName | comment }}
```

Comment:
i just played around a bit, maybe this snippet will help to make the comment function wrap at 80 characters, without splitting words in half.

https://play.golang.org/p/iHXUWlmU2s

Please check i could create a separate PR for this with tests, unless you know a better way to split the comments